open-craft/sprints
# [BB-1603] Split Jira sprints between cells

This introduces actual changes required for managing the sprints separately for each cell. What we've done here:
1. Split the sprint completion task for ending sprints for independently for each cell.
1. Added safety check of the next sprint existence while ending the sprint.
1. Moved out creating new sprint to the Celery task.
1. Moved sprint-specific buttons to `sprint_actions` div, which is rendered by `SprintActionButtons` React component (instead of `CompleteSprints` button).
1. Added new `Create New Sprint` button, which allows any member to create new sprint. This way anybody will be able to plan tickets further than one sprint ahead without asking for manual sprint creation by Sprint Manager. If it will cause problems due to race conditions, we can specify `is_restricted={true}` for this button to restrict its usage to Sprint Managers for semi-automatic sprint creation.

## Testing instructions:
1. Set `.env` according to the ticket description (you'll need to test this with Jira account that has `Sprint Manager` permissions).
1. Start the backend with `docker-compose -f local.yml up --build`.
1. Create superuser with `docker-compose -f local.yml run --rm django python manage.py createsuperuser` (the red button on frontend should be visible for `staff` users only, yellow one should be visible for everyone).
1. Add email to your superuser account under `http://localhost:8000/admin/account/emailaddress/`.
1. Navigate to `frontend` and run `npm i && npm start`.
1. Log into the frontend.
1. Go to the cell's dashboard.
1. Click `Create Next Sprint` button and check that the next sprint has been created, contains the proper cell key, number and date.
